modules/dnn/src/model.cpp

```diff
@@ -637,6 +637,294 @@ void DetectionModel::detect(InputArray frame, CV_OUT std::vector<int>& classIds,
+                // Calculate the resize factor and padding for letterbox
+                resizeFactor = std::min(size.width / (float)Width, size.height / (float)Height);
+                int rh = int(Height * resizeFactor);
+                int rw = int(Width * resizeFactor);
+
+                int top = (size.height - rh) / 2;
+                int bottom = size.height - top - rh;
+                int left = (size.width - rw) / 2;
+                int right = size.width - left - rw;
+
+                box.x = (box.x - left) / resizeFactor;
+                box.y = (box.y - top) / resizeFactor;
+                box.width = (box.width - right) / resizeFactor;
+                box.height = (box.height - bottom)  / resizeFactor;
+            } else {
+                // Handle other cases or throw an error if the padding mode is unsupported
+                CV_Error(Error::StsNotImplemented, "Unsupported padding mode");
+            }
+            // convert from xmin, ymin, xmax, ymax -> xmin, ymin, width, hight
+            box.width = box.width - box.x;
+            box.height = box.height - box.y;
+        }
+    }
+};
+
+YOLODetectionModel::YOLODetectionModel(const String& model, const String& config)
+{
+    impl = makePtr<YOLODetectionModel_Impl>();
+    impl->initNet(readNet(model, config));
+}
+
+YOLODetectionModel::YOLODetectionModel(const String& onnx)
+{
+    impl = makePtr<YOLODetectionModel_Impl>();
+    impl->initNet(readNetFromONNX(onnx));
+}
+
+YOLODetectionModel::YOLODetectionModel()
+{
+    impl = std::static_pointer_cast<Model::Impl>(makePtr<YOLODetectionModel_Impl>());
+}
+
+void YOLODetectionModel::detect(InputArray frame, CV_OUT std::vector<int>& classIds,
+                            CV_OUT std::vector<float>& confidences, CV_OUT std::vector<Rect>& boxes,
+                            float confThreshold, float nmsThreshold){
+
+    CV_Assert(impl != nullptr && impl.dynamicCast<YOLODetectionModel_Impl>() != nullptr); // remove once default constructor is removed
+
+    std::vector<Mat> detections;
+    impl.dynamicCast<YOLODetectionModel_Impl>()->processFrame(frame, detections);
+
+    postProccess(
+        detections,
+        boxes,
+        confidences,
+        classIds,
+        impl->size,
+        confThreshold,
+        nmsThreshold,
+        getNmsAcrossClasses()
+        );
+
+    impl.dynamicCast<YOLODetectionModel_Impl>()->boxRescale(boxes);
+}
+
+void YOLODetectionModel::postProccess(
+    std::vector<Mat>& detections,
+    CV_OUT std::vector<Rect>& keep_boxes,
+    CV_OUT std::vector<float>& keep_confidences,
+    CV_OUT std::vector<int>& keep_classIds,
+    Size inputImgSize,
+    const float confThreshold,
+    const float nmsThreshold,
+    const bool nmsAcrossClasses
+    ){
+
+    bool yolov8 = false;
+    bool darknet = false;
+
+    if (detections[0].dims == 2){
+        darknet = true;
+    }
+
+    if (detections[0].size[1] == 84 and !darknet) {
```

Comment:
It works only for 80 classes. YOLOv8 output shape is `(C+4)xN`. So I recommend compare if `N` is more than `C+4`:
```
detections[0].size[1] < detections[1].size[2]
```

Reply:
Done